sql/sql_acl.cc

```diff
@@ -11166,6 +11166,21 @@ acl_authenticate(THD *thd, uint com_change_user_pkt_len)
       general_log_print(thd, command, (char*) "%s@%s on %s",
                         mpvio.auth_info.user_name, mpvio.auth_info.host_or_ip,
                         mpvio.db.str ? mpvio.db.str : (char*) "");
+
+    if (log_legacy_user && opt_legacy_user_identifier &&
+        strstr(mpvio.auth_info.user_name, opt_legacy_user_identifier))
+    {
+      /* Log the legacy user */
+      char buffer[256];
+      snprintf(buffer, 256, "LEGACY_USER: %s@%s on %s",
```

Comment:
If the database is not assigned for a user, the output string will look like `LEGACY_USER: <user>@<location> on  `.  This looks weird.  One solution, that also avoids the static buffer is this:

```
std::string output= std::string("LEGACY_USER: ") + mpvio.autho_info.user_name +
                   "@" + mpvio.autho_info.host_or_ip;
if (mpvio.db.str != nullptr)
{
  output += std::string(" on ") + mpvio.db.str;
}
...
slow_log_print(thd, output.c_str(), output.size(), &thd->status_var);
...
```